### ISSUE TITLE
## Cross-Mode Alignment Constraint not found in released code

### ISSUE THREAD

**Issue by yifeiliao0908-bit** [Fantasy-AMAP/fantasy-vln#3]
```markdown
## Cross-Mode Alignment Constraint not found in released code

Thank you for the great work! I've been reading your paper and exploring the released code, and I noticed a discrepancy regarding the Cross-Mode Alignment Constraint described in Section 3.5.

### What the paper describes

The paper introduces a Cross-Mode Alignment Constraint with the following formulation (Eq. 7-9):

$$L^*_{Joint} = L_{Align} + L_{CoT}$$

$$L_{Align} = L_{CE}(\hat{A}^T_t, \tilde{A}_t) + L_{CE}(\hat{A}^V_t, \tilde{A}_t) + L_{CE}(\hat{A}^M_t, \tilde{A}_t)$$

where $\tilde{A}_t = sg[\pi_\theta(I, \{o_{\leq t}\}, g_T=0, g_V=0)]$ is the soft target from the non-CoT mode (with stop gradient).

The paper also states (Table 5) that without this alignment constraint, SR drops from 2.44 to 0, suggesting it is essential.

### What the released code implements

In `trainer.py`, `MySeq2SeqTrainer.training_step()` (lines 591-698), the training loop performs 4 separate forward passes (Non-CoT, T-CoT, V-CoT, MM-CoT), each computing a standard CE loss against GT, and returns their mean:

```python
losses = []
for i in range(len(inputs_list)):
    inputs = inputs_list[i]
    loss = self.compute_loss(model, inputs, num_items_in_batch=num_items_in_batch)
    self.accelerator.backward(loss, **kwargs)
    losses.append(loss.detach())
return torch.stack(losses).mean()
```

I searched the entire codebase for keywords like `align`, `alignment`, `soft_target`, `stop_grad`, `detach` (in a distillation context), but could not find:

1. The soft target $\tilde{A}_t$ from the non-CoT mode being used as a teacher signal
2. The stop-gradient (`sg`) mechanism on the non-CoT predictions
3. The explicit $L_{Align}$ term that aligns CoT mode action predictions to non-CoT predictions

The `.detach()` calls in the code appear to be only for logging purposes (`losses.append(loss.detach())`), not for teacher-student distillation.

### My question

It appears that the released code implements cross-mode alignment **implicitly** through shared GT action labels and shared model parameters (multi-branch SFT), rather than the explicit alignment loss described in the paper.

Could you clarify:

1. Is the explicit Cross-Mode Alignment Constraint (with soft targets and stop gradient) implemented in a different version of the code that wasn't released?
2. Or does the released code represent the actual training setup, where alignment is achieved implicitly through shared GT supervision?
3. If the latter, how significant is the gap between the implicit approach (shared GT) and the explicit approach (soft target alignment) in terms of final performance?

Any clarification would be greatly appreciated!
```

**Comment by panda7777777** [comment]
Thank you for your interest in our work and pointing this out. We have updated the code to include the Cross-Mode Alignment Constraint. Please feel free to let us know if you have any further questions.
